lib/archethic/bootstrap.ex

```diff
@@ -206,12 +209,54 @@ defmodule Archethic.Bootstrap do
       Logger.info("Synchronization finished")
     end
 
+    validation_constraints = persist_gen_addr()
     Sync.publish_end_of_sync()
     SelfRepair.start_scheduler()
-
+    Task.await(validation_constraints)
     :persistent_term.put(:archethic_up, :up)
   end
 
+  #  persist gen_addr
+  #  New node joining, sync all txs
+  #  node rejoins and requires sync
+  #  first time network initialization
+  @spec persist_gen_addr :: Task.t()
+  def persist_gen_addr() do
+    # Implemented
+    # :node_shared_secrets, :origin, :oracle , :oracle_summary, :node_rewards ,:mint_rewards
+    Task.Supervisor.async_nolink(Archethic.TaskSupervisor, fn ->
```

Comment:
You don't think to leverage async process for simple persistent_term or ETS inserts

Reply:
my bad, Old commits showed up